Fix C10_UNUSED_DISPATCH_CUDA9_WORKAROUND

pbelevich opened the pull request:
Tested on devfair with the following modules:

- cuda/9.2 cudnn/v7.3-cuda.9.2
- cuda/10.0 cudnn/v7.6-cuda.10.0
- cuda/10.1 cudnn/v7.6.5.32-cuda.10.1
- cuda/10.2 cudnn/v7.6.5.32-cuda.10.2

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37041 Fix C10_UNUSED_DISPATCH_CUDA9_WORKAROUND**

Differential Revision: [D21169790](https://our.internmc.facebook.com/intern/diff/D21169790)

pbelevich committed:
Fix C10_UNUSED_DISPATCH_CUDA9_WORKAROUND

[ghstack-poisoned]

jerryzh168 reviewed (approved):
please make sure it passes CI(both OSS and fb internal) before you land

pbelevich committed:
Update on "Fix C10_UNUSED_DISPATCH_CUDA9_WORKAROUND"


Differential Revision: [D21169790](https://our.internmc.facebook.com/intern/diff/D21169790)

[ghstack-poisoned]

pbelevich committed:
Update on "Fix C10_UNUSED_DISPATCH_CUDA9_WORKAROUND"

Tested on devfair with the following modules:

- cuda/9.2 cudnn/v7.3-cuda.9.2
- cuda/10.0 cudnn/v7.6-cuda.10.0
- cuda/10.1 cudnn/v7.6.5.32-cuda.10.1
- cuda/10.2 cudnn/v7.6.5.32-cuda.10.2


Differential Revision: [D21169790](https://our.internmc.facebook.com/intern/diff/D21169790)

[ghstack-poisoned]
